src/diffpy/utils/diffraction_objects.py

```diff
@@ -186,11 +194,16 @@ def all_arrays(self):
         return self._all_arrays
 
     @all_arrays.setter
-    def all_arrays(self, value):
-        raise AttributeError(
-            "Direct modification of attribute 'all_arrays' is not allowed."
-            "Please use 'insert_scattering_quantity' to modify `all_arrays`."
-        )
+    def all_arrays(self, _):
+        raise AttributeError(_setter_wmsg("all_arrays"))
+
+    @property
```

Comment:
Yes,

```python
    @input_xtype.setter
    def input_xtype(self, _):
        raise AttributeError(_setter_wmsg("input_xtype"))
```

This code basically, gives `AttributeError` when the user attempts to modify `input_xtype` manually.

Reply:
I see.  This is what I am seeing on my screen which is why I thought otherwise, but it may not be showing the latest code on GH?
![image](https://github.com/user-attachments/assets/99d29b01-83f4-4343-b241-68ea2f758a02)

Reply:
not looking far enough down.  Thanks for the explanation